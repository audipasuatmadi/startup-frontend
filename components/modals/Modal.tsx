import React from 'react'
import FocusTrap from 'focus-trap-react'

interface Props {
  children: any
  isLabelOnly?: boolean
}

const Modal = ({children, isLabelOnly}: Props) => {

  
  return (
    <FocusTrap>
      <div className="fixed m-0 left-0 right-0 z-50 w-screen h-screen bg-black opacity-50 flex items-center justify-center" tabIndex={0}>
        {children}
        {isLabelOnly && <div tabIndex={0} /> }
      </div>
    </FocusTrap>
  )
}

export default Modal
