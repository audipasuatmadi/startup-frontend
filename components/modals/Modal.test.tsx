import React from 'react'
import Modal from "./Modal";
import { shallow } from 'enzyme';

describe('Modal render tests', () => {
  it('should render properly', () => {
    const wrapper = shallow(<Modal><p>hello world!</p></Modal>)
    expect(wrapper.find('p').exists()).toBe(true)
    expect(wrapper).toMatchSnapshot()
  })
})