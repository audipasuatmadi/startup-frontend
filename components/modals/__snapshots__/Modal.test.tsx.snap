// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal render tests should render properly 1`] = `
<FocusTrap
  _createFocusTrap={[Function]}
  active={true}
  focusTrapOptions={Object {}}
  paused={false}
>
  <div
    className="fixed m-0 left-0 right-0 z-50 w-screen h-screen bg-black opacity-50 flex items-center justify-center"
    tabIndex={0}
  >
    <p>
      hello world!
    </p>
  </div>
</FocusTrap>
`;
